### ISSUE TITLE
More theme options to enhance readability

### ISSUE THREAD

**Issue by jlacadd** [Vulnerator/Vulnerator#134]
# Prerequisites

- [x ] I have ensured that I am running the latest release
- [x] The issue is repeatable
- [x] The issue has not already been reported
 
# Category

- [ ] Bug
- [x] Enhancement
- [ ] Assistance Request

# Description

## Synopsis
Love the theme option, as I prefer darker backgrounds.  However the "black" or "white" options are a bit limiting.  I find the black background with any available highlight to be difficult to read.  I'd like to see more options for background color and the addition of changing the font color.  For an example of an easier to read dark theme, see the Dark Theme for Powershell ISE (screenshot attached).

## Expected Outcome
- [x] N/A (Non-Bug Issue)

## Actual Outcome
- [x] N/A (Non-Bug Issue)

## Reproduction Steps
- [x] N/A (Non-Bug Issue)

# Attachments

- [ ] Vulnerator Log (Required for bugs) - this can be found at ```C:\Users\%UserName%\AppData\Roaming\Vulnerator\V6Log.txt```, where ```%UserName%``` is the user profile used to run the application
- [x] Screenshots (Optional)
- [ ] Vulnerability Files (_**SANITIZED AND EMAILED**_)
![capture](https://user-images.githubusercontent.com/32691006/31471454-c80a6b2e-af23-11e7-8d7f-84d5d4821787.JPG)

**Comment by amkuchta** [comment]
I'm temporarily throwing this one on hold for a few reasons:

* My backlog is already 35486045640 items long for v6.2.0
* I use pre-defined theme settings from a control library, so I will have to define a custom one to meet this requirement.  Not a huge deal, but time-consuming

I will work to get it incorporated, though!  Thanks for the suggestion, @jlacadd

**Comment by amkuchta** [comment]
@jlacadd I decided I wanted to try this :laughing:

First swags are below - let me know what you think needs to be tweaked (I'm sure I am a little off still)

### "PowerShell Dark"
__*w/ "Cyan" Accent*__
![powershelldark](https://user-images.githubusercontent.com/6255186/31678423-2823c89a-b33c-11e7-803e-86fcf96a17b0.jpg)

### "Matrix"
__*w/ "Lime" Accent*__
![matrix](https://user-images.githubusercontent.com/6255186/31678439-32813caa-b33c-11e7-9ea9-577215992aca.jpg)

**Comment by jlacadd** [comment]
Ah...very nice.  Much improved!  It'll be interesting to see what it looks like with a bunch of data being displayed.

**Comment by amkuchta** [comment]
@jlacadd glad to hear you approve so far.  I'm thinking that I need to darken the background in `PowerShellDark`, and what do you think of the font color?  Should I keep it the light blue, or move over the the ... salmon?... color that was in your screenshot.

@Vulnerator/user-reps, feel free to weigh in, as well.  I was looking through [PowerShell_ISE_Themes](https://github.com/marzme/PowerShell_ISE_Themes), and I think a lot of the styles provided there would translate well to the application - do any of them stick out as "better" to you than the rest?

**Comment by jlacadd** [comment]
Yeah if you want it to be perfect it could a bit darker, but it's pretty close as is.

As for the other color, I think that was more for contrast so the variable names stand out in the code.  For Vulnerator that could come in handy if you want to highlight items in the output or interface, but as a base color for the text it might be a bit much.

The Theme collection is fun, I might try some of them out.  None of them jump out at me as incredible, though I am partial to the dark blue backgrounds as opposed to the pure black, and I kind of like Retta with the orangy text.  Themes are incredibly subjective things...I guess that's why there are so many of them.  I see some guys coding with themes that would make my eyes bleed, but somehow they like it!

**Comment by amkuchta** [comment]
@jlacadd I'm with you, I prefer darker themes (I even have an addon in Chrome that makes GitHub darker when I am at home).  I like dark greys (think "gunmetal" to "charcoal") myself, but like you said, to each their own.

As for the variable name color, I can also create "accents" for the application that would utilize a color like that.  Like I said earlier on, this is my favorite part of the application, so it is something I want to delve into a bit more (especially since I can do it under the guise of providing "accessibility" 😆)

**Comment by amkuchta** [comment]
Updated the "PowerShell Dark" theme using _actual_ hex codes from the screenshot that you gave me (and a red accent this time, because why not?)  I think I have hit on the winning color combination.  Also, I am displaying the "Software" page, which shows how the application will look with data populated

![powershelldark](https://user-images.githubusercontent.com/6255186/31771563-9732b826-b4aa-11e7-8e23-4a41bbbdcb68.jpg)

**Comment by rkotlarz** [comment]
@amkuchta I wouldn't say that any one theme, in particular, would be better over another due to the wide variance in personal preferences from the user base. Given that, how time-consuming would it be to parse an XML theme file and assign it to theme color variables? This would remove the need for you to spend time in the future on additional themes and would allow users to modify the predefined theme settings from a control library to their liking.

Since we're talking about visual usability, it would also be nice to choose the UI scaling to better support 4K monitors. (i.e. 100% -> 300%) I'm assuming that this would be more work that it's currently worth, but after testing out a 4K monitor earlier this year I realized that some of the apps that I use on a weekly basis weren't designed to be operated at 3840 × 2160. Instead, I hoped to go with a 34" ultrawide with a 34" 2560 x 1080 display which worked out better.
